File: EvidenAuctionHouse/src/models/bidCreationDTO.ts
import { Bid } from './bid';

export class BidCreationDTO {
    Bid: Bid;
    ItemPerceivedPrice: number;
    
    constructor(bid: Bid, perceivedPrice: number) {
        this.Bid = bid;
        this.ItemPerceivedPrice = perceivedPrice;
    }
}

File: EvidenAuctionHouseAPI/Models/BidCreationDTO.cs
using DataHandler.Models;
using System.Text.Json.Serialization;

namespace EvidenAuctionHouseAPI.Models
{
    public class BidCreationDTO
    {
        public Bid Bid { get; set; }
        public int ItemPerceivedPrice { get; set; }
        
        // Bezparametrový konstruktor pro JSON deserializaci
        public BidCreationDTO()
        {
        }
        
        // Konstruktor s parametry pro ruční vytváření
        [JsonConstructor]
        public BidCreationDTO(Bid bid, int itemPerceivedPrice)
        {
            Bid = bid;
            ItemPerceivedPrice = itemPerceivedPrice;
        }
    }
}
